### DISCUSSION TITLE
Multiple entry point in android

### DISCUSSION THREAD

**Discussion by vishva238** [beeware/toga#1311] [Q&A]
I would like to know the process of multi activity application for android.

Like Starting getting an SEND intent from other apps and use it in my application.

**Comment by vishva238** [comment]
> I would like to know the process of multi activity application for android.
> 
> Like Starting getting an SEND intent from other apps and use it in my application.

I'm asking about intent handling without main activity.

**Reply by freakboy3742** [reply]
At present, we don't have built-in support for receiving intents, or for multi-activity apps. 

The concept of Intents and Activities are very Android-specific; it's unlikely we're ever going to have a *direct* mapping of those features in Toga, as Toga is a cross platform toolkit. We might, however, add an API to enable a Toga app to register itself as a handler for a particular file/URL type. 

That said - the BeeWare stack is deliberately layered; and you can mix-and-match layers. Toga is the abstracted cross-platform layer; but if you go one layer deeper to the Rubicon layer, you can talk directly to Android APIs. You'll also need to manually manage the interaction between the Android Gradle project that represents those activities or intents. Briefcase will give you a template that supports a single Activity; but if you want more than one activity, you'll need to add that support yourself.

Unfortunately, I can't point you at a simple HOWTO guide for any of this, though; you'll need to piece it together from the bits that are already there.

**Reply by vishva238** [reply]
Ok I register through Androidmanifest file. 
Is it possible to edit the java file created before building the apk
And invoke python from that java function

Or Any other way like starting the main windows and search if there is any intent  using Rubicon and then continue with preferred windows
Is it possible
If so can I get a short guide

**Reply by freakboy3742** [reply]
It's definitely possible to edit the Java file before building the APK; once the project has been generated with Briefcase, you can edit whatever you want. If you have changes you want to make permanent, it might be worth creating a custom briefcase template that incorporates those changes.

Is it possible to invoke Python from a java function? Yes- although the dependency is technically reversed. You define a Java interface, write a Python class that implements that interface, expose a Java endpoint that lets you set an object that implements the Java interface, and use Python code to call that Java endpoint to set the object to your Python instance. That's exactly what the Python MainActivity does when it starts up in a normal Briefcase project; if you want to do a similar thing with a different Activity, then you need to duplicate the concept.

However, as I said in my last response, I can't point you at any documentation for this; you'll need to piece this together from what is currently there.

**Reply by vishva238** [reply]
Ok :) Thank You so much for the information. 
I'll work in that and will update.

**Reply by NadaComa** [reply]
@vishva238 Hi just wandering if you made any progress and could provide a small guide on what to change.
I'd like to be able to send URLs to my app.
